### PR TITLE
Notifier Microservice

### PR DESCRIPTION
**Pull request by ives1227**
**A microservice that reads messages from a queue and sends out an email*
* * *

**Ticket Link**: https://jira.huit.harvard.edu/browse/LTSEPADD-94 (Parent is https://jira.huit.harvard.edu/browse/LTSEPADD-93)

# How should this be tested?

Follow the instructions here: https://github.com/harvard-lts/dais-notifier/blob/LTSEPADD-93/README.md#testing

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? yes
- integration tests? yes

### CODE REVIEW

File: gunicorn.conf.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        "console": {
+            "class": "logging.FileHandler",
+            "formatter": "json_formatter",
+            "filename": f"/home/appuser/logs/console_{container_id}_{timestamp}.log",
+            "mode": "a"
+        }
+    },
+    "loggers": {
+        'gunicorn.error': {
+            'handlers': ['console'],
```
Comment (awoods) [inline gunicorn.conf.py:55]: Should this handler instead be: 'error_console'?

File: tests/integration/mqutils.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import logging
+import os
+
+import stomp
+
+class ConnectionParams:
```
Comment (awoods) [inline tests/integration/mqutils.py:6]: Is this class ever used?

File: tests/unit/test_notifier.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import pytest, sys
+sys.path.append('app')
+from notifier.dais_notifier import SmtpMailingService 
+
+   
+def test_send_email_with_no_recipiants():
```
Comment (awoods) [inline tests/unit/test_notifier.py:6]: Typo: 'recipiants'

File: tests/unit/test_notifier.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    subject = "Test DAIS Notification Email"
+    body = "Test DAIS Notification Email\nBody"
+    
+    mailing_service = SmtpMailingService()
+    retval = mailing_service.send_email(subject, body)
+    
+    #Check that the email was sent
+    assert retval
+
+def test_send_email_with_one_recipiant():
```
Comment (awoods) [inline tests/unit/test_notifier.py:17]: Typo: 'recipiants'

File: tests/unit/test_notifier.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    subject = "Test DAIS Notification Email"
+    body = "Test DAIS Notification Email\nBody"
+    recipients = "valdeva_crema@harvard.edu"
+    mailing_service = SmtpMailingService()
+    retval = mailing_service.send_email(subject, body, recipients = None)
+    
+    #Check that the email was sent
+    assert retval
+    
+def test_send_email_with_two_recipiants():
```
Comment (awoods) [inline tests/unit/test_notifier.py:28]: Typo: 'recipiants'

**Review by awoods — CHANGES_REQUESTED** [review]
Some minor comments.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,49 @@
+FROM python:3.8-slim-buster
```
[the remark this answers is not in the document]
Comment (ives1227) [reply Dockerfile]: I updated it - an oversight

File: tests/integration/mqutils.py (hunk 2 above)
Comment (ives1227) [reply tests/integration/mqutils.py:6]: Yes, throughout the mqutils module.

File: tests/unit/test_notifier.py (hunk 5 above)
Comment (ives1227) [reply tests/unit/test_notifier.py]: fixed

File: tests/unit/test_notifier.py (hunk 4 above)
Comment (ives1227) [reply tests/unit/test_notifier.py]: fixed

File: gunicorn.conf.py (hunk 1 above)
Comment (awoods) [reply gunicorn.conf.py:55]: Did you see this comment?

**Review by awoods — CHANGES_REQUESTED** [review]

**Comment by ives1227** [comment]
@awoods just fixed that typo.

File: gunicorn.conf.py (hunk 1 above)
Comment (ives1227) [reply gunicorn.conf.py]: Just fixed.

**Review by awoods — APPROVED** [review]

**Pull request merged**